includes/admin/importers/class-wc-product-csv-importer-controller.php

```diff
@@ -8,83 +8,180 @@
+	 * Get the URL for the next step's screen.
+	 * @param string step   slug (default: current step)
+	 * @return string       URL for next step if a next step exists.
+	 *                      Admin URL if it's the last step.
+	 *                      Empty string on failure.
 	 */
-	public function dispatch() {
-		include_once( WC_ABSPATH . 'includes/import/class-wc-product-csv-importer.php' );
+	public function get_next_step_link( $step = '' ) {
+		if ( ! $step ) {
+			$step = $this->step;
+		}
 
-		$this->header();
+		$keys = array_keys( $this->steps );
 
-		$step = empty( $_GET['step'] ) ? 0 : (int) $_GET['step'];
+		if ( end( $keys ) === $step ) {
+			return admin_url();
+		}
 
-		switch ( $step ) {
+		$step_index = array_search( $step, $keys );
 
-			case 0:
-				$this->greet();
-				break;
+		if ( false === $step_index ) {
+			return '';
+		}
 
-			case 1 :
-				check_admin_referer( 'import-upload' );
+		$params = array(
+			'step'      => $keys[ $step_index + 1 ],
+			'file'      => $this->file,
+			'file_url'  => $this->file_url,
+			'delimiter' => $this->delimiter,
+			'_wpnonce'  => wp_create_nonce( 'woocommerce-csv-importer' ), // wp_nonce_url() escapes & to &amp; breaking redirects.
+		);
 
-				if ( $this->handle_upload() ) {
-					if ( $this->id ) {
-						$file = get_attached_file( $this->id );
-					} else {
-						$file = ABSPATH . $this->file_url;
-					}
+		return add_query_arg( $params );
+	}
 
-					$this->importer_mapping( $file );
-				}
-				break;
+	/**
+	 * Output header view.
+	 */
+	protected function output_header() {
+		include( dirname( __FILE__ ) . '/views/html-csv-import-header.php' );
+	}
+
+	/**
+	 * Output steps view.
+	 */
+	protected function output_steps() {
+		include( dirname( __FILE__ ) . '/views/html-csv-import-steps.php' );
+	}
+
+	/**
+	 * Output footer view.
+	 */
+	protected function output_footer() {
+		include( dirname( __FILE__ ) . '/views/html-csv-import-footer.php' );
+	}
+
+	/**
+	 * Get importer instance.
+	 *
+	 * @param  string $file File to import.
+	 * @param  array  $args Importer arguments.
+	 * @return WC_Product_CSV_Importer
+	 */
+	protected function get_importer( $file, $args = array() ) {
+		$importer_class = apply_filters( 'woocommerce_product_csv_importer_class', 'WC_Product_CSV_Importer' );
+		return new $importer_class( $file, $args );
+	}
 
+	/**
+	 * Add error message.
+	 */
+	protected function add_error( $error ) {
+		$this->errors[] = $error;
+	}
+
+	/**
+	 * Add error message.
+	 */
+	protected function output_errors() {
+		if ( $this->errors ) {
+			foreach ( $this->errors as $error ) {
+				echo '<p class="error inline">' . esc_html( $error ) . '</p>';
+			}
+		}
+	}
+
+	/**
+	 * Dispatch current step and show correct view.
+	 */
+	public function dispatch() {
+		if ( ! empty( $_POST['save_step'] ) && isset( $this->steps[ $this->step ]['handler'] ) ) {
+			call_user_func( $this->steps[ $this->step ]['handler'], $this );
```

Comment:
This line threw the following error when I ran the importer:

`
Warning: call_user_func() expects parameter 1 to be a valid callback, function '' not found or invalid function name in /srv/www/wordpress-default/public_html/wp-content/plugins/woocommerce/includes/admin/importers/class-wc-product-csv-importer-controller.php on line 176
`

I think if we change the `isset` to `! empty` we should be good.